pkg/app/app.go

```diff
@@ -88,6 +88,19 @@ func defaultFuncApiUrl(config configuration.Configuration, logger *zerolog.Logge
 
 		if len(urlFromOauthToken) > 0 && len(urlFromOauthToken[0]) > 0 {
```

Comment:
suggestion: From a UX perspective I'm a bit unhappy with the logical flow here, as the warning on L86 (can't link it 🤷🏻 ) will always show up in logs of customers using PATs. The general fallback logic seems to not really favor early returns or breaks. Is it possible to refactor this a bit to avoid warning-spams?

Also it seems to just continue even if the PAT-part fails, causing an error further down the execution tree. I wonder if an earlier authentication error would be nicer and cause easier debugging for us in the future.